tools/restatectl/src/commands/log/reconfigure.rs

```diff
@@ -0,0 +1,165 @@
+    provider: ProviderKind,
+
+    /// Replication property
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long)]
+    replication_factor: Option<NonZeroU8>,
+    /// A comma-separated list of the nodes in the nodeset. e.g. N1,N2,N4 or 1,2,3
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long, value_delimiter=',', num_args = 0..)]
+    nodeset: Vec<PlainNodeId>,
+    /// The generational node id of the sequencer node, e.g. N1:1
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long)]
+    sequencer: Option<GenerationalNodeId>,
+}
+
+async fn reconfigure(connection: &ConnectionInfo, opts: &ReconfigureOpts) -> anyhow::Result<()> {
+    let channel = grpc_connect(connection.cluster_controller.clone())
+        .await
+        .with_context(|| {
+            format!(
+                "cannot connect to cluster controller at {}",
+                connection.cluster_controller
+            )
+        })?;
+
+    let mut client =
+        ClusterCtrlSvcClient::new(channel).accept_compressed(CompressionEncoding::Gzip);
+
+    let (provider, params) = match opts.provider {
+        ProviderKind::Local => (ProtoProviderKind::Local, rand::random::<u64>().to_string()),
+        #[cfg(any(test, feature = "memory-loglet"))]
+        ProviderKind::InMemory => (ProtoProviderKind::InMemory, String::default()),
+        #[cfg(feature = "replicated-loglet")]
+        ProviderKind::Replicated => (
+            ProtoProviderKind::Replicated,
+            replicated_loglet_params(&mut client, opts).await?,
+        ),
+        #[allow(unreachable_patterns)]
+        _ => anyhow::bail!("Provider kind is not supported"),
+    };
+
+    client
+        .seal_and_extend_chain(SealAndExtendChainRequest {
+            log_id: opts.log_id,
+            min_version: Some(Version {
+                value: opts.min_version.get(),
+            }),
+            provider: provider.into(),
+            segment_index: opts.segment_index,
+            params,
+        })
+        .await?;
+
+    Ok(())
+}
+
+async fn replicated_loglet_params(
+    client: &mut ClusterCtrlSvcClient<Channel>,
+    opts: &ReconfigureOpts,
+) -> anyhow::Result<String> {
+    let mut logs_resposne = client
+        .list_logs(ListLogsRequest {})
+        .await
+        .context("Failed to get logs metadata")?
+        .into_inner();
+
+    let logs = StorageCodec::decode::<Logs, _>(&mut logs_resposne.logs)?;
+    let chain = logs
+        .chain(&LogId::from(opts.log_id))
+        .with_context(|| format!("Unknown log id '{}'", opts.log_id))?;
+
+    let tail_index = opts.segment_index.unwrap_or(chain.tail_index().into());
+
+    // format is, log_id in the higher order u32, and segment_index in the lower
+    let loglet_id: u64 =
```

Comment:
This needs to align with what cluster controller does. Cluster controller at the moment generates random u64.

Reply:
Or piggyback on the LogletProvider trait